s2a/src/main/java/io/grpc/s2a/channel/S2AHandshakerServiceChannel.java

```diff
@@ -178,16 +162,11 @@ public <ReqT, RespT> ClientCall<ReqT, RespT> newCall(
     @SuppressWarnings("FutureReturnValueIgnored")
     public void close() {
       delegate.shutdownNow();
-      boolean isDelegateTerminated;
       try {
-        isDelegateTerminated =
-            delegate.awaitTermination(DELEGATE_TERMINATION_TIMEOUT.getSeconds(), SECONDS);
+        delegate.awaitTermination(CHANNEL_SHUTDOWN_TIMEOUT.getSeconds(), SECONDS);
       } catch (InterruptedException e) {
-        isDelegateTerminated = false;
+        Thread.currentThread().interrupt();
```

Comment:
if `shutdownNow` didn't complete, you'll probably want to know about it for debugging, so I'd suggest logging a warning here.

Reply:
Done in [653ac1e](https://github.com/grpc/grpc-java/pull/11539/commits/653ac1e8e415eda8df7c941b1e91296c3ff2eb9b)